removes Ruby (#2251)
temporalio/documentation: docs/dev-guide/index.mdx

> The following SDKs do not yet have developer's guides:

Reviewer: ```suggestion
The following SDK does not yet have a developer's guide:
```
